frontendExternalSystem/src/main/java/com/virtuslab/gitmachete/frontend/externalsystem/project/MacheteProjectResolver.java

```diff
@@ -28,16 +24,6 @@
       @Nullable MacheteExecutionSettings settings,
       ExternalSystemTaskNotificationListener listener)
       throws ExternalSystemException, IllegalArgumentException, IllegalStateException {
-
-    var graphTableProvider = Option.of(settings)
```

Comment:
Wow okay... any idea what was the original purpose of this piece? As in, why it used to be needed back then but redundant/erroneous now?...

Reply:
The point being, whether by removing this stuff we don't break sth else

Reply:
Now we have got this part of the code

```
addAncestorListener(new AncestorListenerAdapter() {
      @Override
      public void ancestorAdded(AncestorEvent event) {
        var gitRepository = selectedGitRepositoryProvider.getSelectedGitRepository().getOrNull();
        if (gitRepository != null) {
          var macheteFilePath = getMacheteFilePath(gitRepository);
          Runnable queueDiscoverOperation = () -> graphTable.queueDiscover(macheteFilePath, () -> {});
          var rediscoverSuggester = new RediscoverSuggester(gitRepository, queueDiscoverOperation);
          graphTable.queueRepositoryUpdateAndModelRefresh(() -> rediscoverSuggester.perform());
        }
      }
    });
```

https://github.com/VirtusLab/git-machete-intellij-plugin/blob/59f1e2000aa390475f2a50005f2acf064bb0f745/frontendUiRootImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/GitMachetePanel.java#L60

The reason is that we want to queue repository (first) update at the right moment. Resolver called it at the very beginning of the IDE run. In the meantime came the above part which calls the update later, on Git Machete tab opening.